ndbench-core/src/main/java/com/netflix/ndbench/core/util/ConstantStepWiseRateIncreaser.java

```diff
@@ -0,0 +1,132 @@
+
+    private final int indexOfLastInTable;
+    private final ImmutableList<Cell> clockTimeToRateTable;
+
+    @Immutable
+    private class Cell {
+        private final long time;
+        private final double rate;
+
+        Cell(long time, double rate) {
+            this.time = time;
+            this.rate = rate;
+        }
+    }
+
+
+    /**
+     * Returns a step-wise rate increaser which will ramp from 'initRate' to 'finalRate' over the course of
+     * 'incrementIntervalMillisecs'.  The number of steps by which the rate increases will be determined by
+     * the value rampPeriodMillisecs / incrementIntervalMillisecs (which MUST evaluate to an integral value with
+     * no remainder). At each step the rate will increase constantly by (finalRate - initRate)  / number-of-steps.
+     */
+    public ConstantStepWiseRateIncreaser(int rampPeriodMillisecs,
+                                  int incrementIntervalMillisecs,
+                                  int initRate,
+                                  int finalRate) {
+        if (!(initRate >= 0)) {
+            throw new IllegalArgumentException("initRate must be >= 0");
+        }
+        if (!(finalRate > 0)) {
+            throw new IllegalArgumentException("finalRate must be > 0");
+        }
+        if (!(finalRate > initRate)) {
+            throw new IllegalArgumentException("finalRate must be > initRate");
+        }
+        if (!(rampPeriodMillisecs > 0)) {
+            throw new IllegalArgumentException("rampPeriodMillisecs must be > 0");
+        }
+        if (!(incrementIntervalMillisecs > 0)) {
+            throw new IllegalArgumentException("incrementIntervalMillisecs must be > 0");
+        }
+        if (rampPeriodMillisecs % incrementIntervalMillisecs != 0) {
+            throw new IllegalArgumentException(
+                    "rampPeriodMillisecs should be evenly divisible by incrementIntervalMillisecs");
+        }
+        if (rampPeriodMillisecs / incrementIntervalMillisecs > MAX_STEPS) {
+            throw new IllegalArgumentException(
+                    "rampPeriodMillisecs / incrementIntervalMillisecs should not exceed MAX_STEPS (" + MAX_STEPS + ")");
+        }
+
+
+        final ArrayList<Cell> tmpClockTimeToRateTable = new ArrayList<Cell>();
+
+        int numSteps = rampPeriodMillisecs / incrementIntervalMillisecs;
+        double spread = (finalRate - initRate) * 1.0;
+        double rateIncrementPerStep = spread / numSteps;
+
+        for (int i = 0; i <= numSteps; i++) {
+            tmpClockTimeToRateTable.add(
+                    new Cell(
+                            i * incrementIntervalMillisecs,
+                            initRate + (i * rateIncrementPerStep)
+                    )
+            );
+        }
+
+        // initialize high values sentinel cell
+        tmpClockTimeToRateTable.add(new Cell(Long.MAX_VALUE, 0 /* dummy never used for rate */));
+
+        indexOfLastInTable = numSteps + 1;
```

Comment:
Why maintain a separate variable to store list size? Why not query `clockTimeToRateTable.size()`? This would be less error prone.